.github/workflows/code-validation.yaml

```
@@ -0,0 +1,58 @@
+name: Code Validation
+
+on:
+  push:
+    branches:
+      - '**'
+
+jobs:
+  validation:
+    name: Tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        test: [style, unit, csi, controller]
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version-file: 'go.mod'
+        cache: true
+
+    - name: Code formatting check
+      if: matrix.test == 'style'
+      run: make check_style
+
+    - name: Unit tests
+      if: matrix.test == 'unit'
+      run: make test-unit
+
+    - name: CSI spec compliance tests
+      if: matrix.test == 'csi'
+      run: make test-csi-compliance
+
+    - name: Controller tests (K8s API mock)
+      if: matrix.test == 'controller'
+      run: make controller-integration-test
```

if you name your each case according to the makefile rule, you could:
```suggestion
    - name: Run ${{ matrix.test }}
      run: make ${{ matrix.test }}
```

---

pretty cool, pretty cool 🫶